packages/types/src/translators/action-view.test.ts

```diff
@@ -0,0 +1,326 @@
+import { ActionView } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/transaction/v1alpha1/transaction_pb';
+import { asPublicActionView, asReceiverActionView } from './action-view';
+import { describe, expect, test, vi } from 'vitest';
+
+describe('asPublicActionView()', () => {
+  describe('when passed `undefined`', () => {
+    test('returns an empty action view', () => {
+      expect(asPublicActionView(undefined).equals(new ActionView())).toBe(true);
+    });
+  });
+
+  describe('when passed an action view with an `undefined` case', () => {
+    test('returns an empty action view', () => {
+      expect(
+        asPublicActionView(new ActionView({ actionView: { case: undefined } })).equals(
+          new ActionView(),
+        ),
+      ).toBe(true);
+    });
+  });
+
+  describe('when passed a spend action view', () => {
+    const actionView = new ActionView({
```

Comment:
These view definitions get massive in these test files. In the future, I'd like to integrate with a library like [`factory.ts`](https://github.com/willryan/factory.ts) to more easily build objects for various use cases, but this PR is already bloated enough without that addition :D

Reply:
Ah yes. A factory or some other way to generate this data more sanely would be nice.